ee/tabby-webserver/src/service/github_repository_provider.rs

```diff
@@ -184,4 +203,75 @@ mod tests {
         assert_eq!(repos.len(), 1);
         assert!(repos[0].active);
     }
+
+    #[tokio::test]
+    async fn test_github_repository_provider_crud() {
+        let db = DbConn::new_in_memory().await.unwrap();
+        let service = new_github_repository_provider_service(db);
+
+        let id = service
+            .create_github_repository_provider("example".into(), "id".into(), "secret".into())
+            .await
+            .unwrap();
+
+        // Test retrieving github provider by ID
+        let provider1 = service
+            .get_github_repository_provider(id.clone())
+            .await
+            .unwrap();
+        assert_eq!(
```

Comment:
how about use assert_match! instead of using partial equal?

Reply:
`assert_matches!` is nightly: https://doc.rust-lang.org/std/assert_matches/macro.assert_matches.html

Reply:
Tried this and it's not really good for this situation, where you want to check exact equality of a lot of fields. Pattern matching wants to extract the values, not compare them in-place.
![image](https://github.com/TabbyML/tabby/assets/14198267/3add0da9-61a1-45c3-815d-9832d701e008)
![image](https://github.com/TabbyML/tabby/assets/14198267/94c2baea-7314-4cda-b52c-a36ef01c7619)